### PR TITLE
[Fixit] Allow patching labels of services and plans

### PR DESCRIPTION
**Pull request by DimitarPetrov**
## Motivation

Currently there is no way to add/remove labels on service offerings and service plans.

## Approach

Introduces a patch API to service offerings and service plans apis that allows patching only the labels.

### CODE REVIEW

File: test/service_offering_test/service_offering_test.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ func TestServiceOfferings(t *testing.T) {
 
 var _ = test.DescribeTestsFor(test.TestCase{
 	API:            "/v1/service_offerings",
-	SupportsLabels: false,
+	SupportsLabels: true,
```
Comment (dpanayotov) [inline test/service_offering_test/service_offering_test.go]: As everything now supports labels, this field and the checks can be removed completely

File: api/base_controller.go [hunk 2]
```diff
@@ -236,6 +236,33 @@ func (c *BaseController) PatchObject(r *web.Request) (*web.Response, error) {
 	return util.NewJSONResponse(http.StatusOK, object)
 }
 
+// PatchObject handles the update of labels of the object with the id specified in the request
+func (c *BaseController) PatchLabelsOnly(r *web.Request) (*web.Response, error) {
```
Comment (dpanayotov) [inline api/base_controller.go]: this is a lot of code duplication. Can you see how to extract this part?

File: api/base_controller.go [hunk 3]
```diff
@@ -236,6 +236,33 @@ func (c *BaseController) PatchObject(r *web.Request) (*web.Response, error) {
...
+		return nil, util.HandleStorageError(err, string(c.objectType))
+	}
+
+	labels, _, _ := query.ApplyLabelChangesToLabels(labelChanges, objFromDB.GetLabels())
+	objFromDB.SetLabels(labels)
+
+	object, err := c.repository.Update(ctx, objFromDB, labelChanges...)
+	if err != nil {
+		return nil, util.HandleStorageError(err, string(c.objectType))
+	}
```
Comment (dpanayotov) [inline api/base_controller.go]: credentials stripping is missing

File: api/filters/patch_labels_only.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+		{
+			Matchers: []web.Matcher{
+				web.Path(web.ServicePlansURL + "/**"),
+				web.Methods(http.MethodPatch),
+			},
+		},
+	}
+}
+
+func onlyLabelsRequest(body []byte) bool {
```
Comment (KirilKabakchiev) [inline api/filters/patch_labels_only.go]: you could inline this function - its short enough and is not used anywhere else

**Pull request merged**